internal/config/profile.go

```diff
@@ -129,15 +130,18 @@ func (p *profile) SetName(name string) {
 
 func Set(name, value string) { p.Set(name, value) }
 func (p *profile) Set(name, value string) {
-	viper.Set(fmt.Sprintf("%s.%s", p.name, name), value)
+	settings := viper.GetStringMapString(p.Name())
+	settings[name] = value
+	viper.Set(p.name, settings)
```

Comment:
this was driving my mad, I had no idea why so many tests started failing, turns out if you set a key with viper, viper do not merge, it will overwrite and since we were doing dots like `default.output` it would overwrite all default to just be `default.output` this also leads me to believe we may have had some unknown bug around with this